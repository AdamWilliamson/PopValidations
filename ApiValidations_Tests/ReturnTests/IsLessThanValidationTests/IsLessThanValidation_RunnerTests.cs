using ApprovalTests;
using FluentAssertions;
using ApiValidations;
using ApiValidations_Tests.TestHelpers;

namespace PopValidations_Tests.returnTests.IsLessThanValidationTests;

public class IsLessThanApi
{
    public int Set() { return 1; }
}

public class IsLessThan_TestingValidator : ApiValidator<IsLessThanApi>
{
    public IsLessThan_TestingValidator()
    {
        DescribeFunc(x => x.Set()).Return.IsLessThan(int.MaxValue);
    }
}

public class IsLessThanValidation_RunnerTests
{
    [Fact]
    public void WhenDescribing_ItReturnsTheValidation()
    {
        // Arrange
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new IsLessThan_TestingValidator());

        // Act
        var description = runner.Describe();

        // Assert
        description.Results.Should().HaveCount(1);
        description.Results.Should().HaveCount(ValidatableHelper.GetValidatableCount<IsLessThanApi>(ValidatableType.NoExceptions));
        Approvals.VerifyJson(JsonConverter.ToJson(description));
    }
}
