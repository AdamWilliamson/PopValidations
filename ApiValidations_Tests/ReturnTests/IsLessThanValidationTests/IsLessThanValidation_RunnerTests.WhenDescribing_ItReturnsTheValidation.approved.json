{
    "Results": [
        {
            "Property": "Set()->Int32:Return(Int32)",
            "Outcomes": [
                {
                    "Validator": "IsLessThanValidation",
                    "Message": "Must be less than '2147483647'.",
                    "Values": [
                        {
                            "Key": "value",
                            "Value": "2147483647"
                        }
                    ]
                }
            ],
            "ValidationGroups": []
        }
    ]
}